### ISSUE TITLE
Confusion about learning rate and batch_size when using horovod

### ISSUE THREAD

**Issue by Queequeg92** [uber/horovod#241]
When I transfer a tensorflow single-gpu code to horovod, should I adjust the learning rate and the batch_size?

**Comment by alsrgv** [comment]
@Queequeg92, you don't have to adjust batch size, but you generally need to adjust learning rate.  Facebook published [ImageNet in 1hr](https://research.fb.com/wp-content/uploads/2017/06/imagenet1kin1h5.pdf) paper demonstrating linear scaling. If you're using Keras, we have it implemented as a plug-in, e.g. see [ImageNet training example](https://github.com/uber/horovod/blob/master/examples/keras_imagenet_resnet50.py).

**Comment by Queequeg92** [comment]
Thanks for your reply.
Does horovod change the effective batch size? If the batch size of a tensorflow single-gpu code is 128 and it's distribute to 4 gpus by horovod, dose the effective batch size change to 128*4? I just want to keep the effective batch size and learning rate schedule unchanged.

**Comment by alsrgv** [comment]
@Queequeg92, that's going to be tricky.  Yes, if you keep batch size per GPU the same, effective batch size will grow according to your formula.  You can decrease batch size per GPU, but GPUs are not optimized for small batch processing, and pretty quickly scaling will not increase performance anymore.  Hence, industry is trying to figure out how to do training effectively with large batches.

**Comment by Queequeg92** [comment]
@alsrgv Got it! :clap::clap::clap:Large batches make training effectively. But for people doing research, some models are too large for a single gpu system. What they want to do is to distribute large models to a multi-gpu system and keep the effective batch size unchanged. I think the docs and examples should make the effective batch size part more clear. So people could adjust the effective batch size based on different applications (industry or research):smile_cat:.

**Comment by alsrgv** [comment]
@Queequeg92, good point - we'll update the docs.
